backend/experimental_unixcoder/bug_localization.py

```diff
@@ -1,81 +1,112 @@
+            except Exception as e:
+                print(f"Error processing chunk {i // chunk_size + 1}: {e}")
+                continue
+
         return embeddings
 
+
     # File Ranking for Bug Localization
-    def rank_files(self, query_embedding, db_embeddings):
+    def rank_files(self, query_embeddings, db_embeddings):
         """
-        Ranks files based on similarity to the query embedding.
-        
+        Ranks files based on similarity to the query embeddings.
+
         Parameters:
-        - query_embedding: The embedding of the bug report or query.
-        - db_embeddings: A list of tuples, where each tuple contains (file_id, embedding)
-                         representing the file's unique identifier and its precomputed embedding.
-                         
+        - query_embeddings: A list of embeddings (as lists) for the query (bug report).
+        - db_embeddings: A list of tuples, where each tuple contains (file_id, embeddings)
+                         where embeddings is a list of embeddings (as lists) for that file.
+
         Returns:
-        - A sorted list of (file_id, similarity_score) tuples in descending order of similarity.
+        - A sorted list of (file_id, max_similarity_score) tuples in descending order of similarity.
         """
-        # Normalize the query embedding
-        norm_query_embedding = torch.nn.functional.normalize(query_embedding, p=2, dim=1)
-        
-        # Calculate similarity scores
         similarities = []
-        for file_id, embedding in db_embeddings:
-            # Ensure db_embedding is on the same device
-            db_embedding = embedding.to(self.device)
-            similarity = torch.einsum("ac,bc->ab", norm_query_embedding, db_embedding).item()
-            similarities.append((file_id, similarity))
-        
-        # Sort files by similarity score in descending order
+
+        for file_id, file_embeddings in db_embeddings:
+            max_similarity = float('-inf')
+
+            # Convert query embeddings and file embeddings from lists back to tensors
+            for query_embedding in query_embeddings:
+                query_tensor = torch.tensor(query_embedding, device=self.device)
+                for file_embedding in file_embeddings:
+                    file_tensor = torch.tensor(file_embedding, device=self.device)
+
+                    # Compute similarity
+                    similarity = torch.nn.functional.cosine_similarity(
+                        query_tensor, file_tensor, dim=1
+                    ).item()
+
+                    if similarity > max_similarity:
+                        max_similarity = similarity
+
+            similarities.append((file_id, max_similarity))
+
         similarities.sort(key=lambda x: x[1], reverse=True)
         return similarities
 
-# Main function that runs if the script is executed directly
 if __name__ == "__main__":
```

Comment:
nit: We should move this into a unit testing script to remove the bloat from this script.